### ISSUE TITLE
Additional dataset issue

### ISSUE THREAD

**Issue by tlscabinet** [KevinMusgrave/pytorch-metric-learning#320]
Hello, for now on I have some specific issue related to computing metrics during training and validation steps.
First I have to describe a little bit the training process.
I have my train and test datasets which were generated from original dataset in such way that each sample of original dataset was splitted into n new smaller objects. 
Now during the training I want to train on augmented dataset and compute also the embeddings of original dataset to compare how close my augmented data to original objects, I didn't find the way how to integrate in into training procedure with help of AccuracyCalculator because basically it takes as input the embeddings, so can you tell me where I have to modify the code to achieve my goal?

P.S. thank you for continuous help and this package

**Comment by KevinMusgrave** [comment]
Put the original dataset in ```dataset_dict```, and set ```splits_to_eval``` so that the original dataset is the reference dataset:

```python
dataset_dict = {"val": val_dataset, "original": original_dataset}

# For k-nn accuracy metrics, 
# this will find the nearest neighbors for samples in "val" 
# using "original" as the search space
splits_to_eval = [("val", ["original"])]

end_of_epoch_hook = hooks.end_of_epoch_hook(tester, 
                                            dataset_dict, 
                                            model_folder, 
                                            test_interval = 1,
                                            patience = 1,
                                            splits_to_eval = splits_to_eval)
```

**Comment by tlscabinet** [comment]
But is there any way to pass few like  splits_to_eval = [("val",
["original_test"]), ('train', ['train']), ("train", ["original_train"]),
now it gives the error

**Comment by KevinMusgrave** [comment]
Yes your example is valid, except that you have the "train" query twice, which isn't yet supported (see https://github.com/KevinMusgrave/pytorch-metric-learning/issues/252).

To get around this limitation, you can use 2 different keys for the train set:

```python
dataset_dict = {"train": train_dataset, "train2": train_dataset, "original_train": original_train_dataset}

splits_to_eval = [('train', ['train']), ("train2", ["original_train"])]
```

**Comment by tlscabinet** [comment]
got it, thank you!!
